packages/grid/x-data-grid-pro/src/utils/tree/utils.ts

```diff
@@ -235,3 +237,45 @@ export const createUpdatedGroupsManager = (): GridRowTreeUpdatedGroupsManager =>
     this.value[groupId][action] = true;
   },
 });
+
+export const getVisibleRowsLookup = ({
+  tree,
+  filteredRowsLookup,
+}: {
+  tree: GridRowsState['tree'];
+  filteredRowsLookup: GridFilterState['filteredRowsLookup'];
+}) => {
+  if (!filteredRowsLookup) {
+    return {};
+  }
+
+  const visibleRowsLookup: Record<GridRowId, boolean> = {};
+
+  const handleTreeNode = (node: GridTreeNode, areAncestorsExpanded: boolean) => {
+    const isPassingFiltering = filteredRowsLookup[node.id];
+
+    if (node.type === 'group') {
+      node.children.forEach((childId) => {
+        const childNode = tree[childId];
+        handleTreeNode(childNode, areAncestorsExpanded && !!node.childrenExpanded);
+      });
+    }
+
+    visibleRowsLookup[node.id] = isPassingFiltering && areAncestorsExpanded;
+
+    // TODO rows v6: Should we keep storing the visibility status of footer independently or rely on the group visibility in the selector ?
```

Comment:
Do you have any opinions on this comment? Are footers sometimes hidden?

Reply:
Yes, you can have aggregation footer for each row group - see https://mui.com/x/react-data-grid/aggregation/#AggregationGetAggregationPosition.tsx

When the group is filtered out, the aggregation footer should be hidden as well.

Reply:
IIUC, the visibility of the group is always equal to the visibility of the footer, right?

Reply:
Not exactly, you can have a visible row group, but if it's not expanded - the footer row shouldn't be visible:
https://github.com/mui/mui-x/blob/9aa0d401bcf80cda9f7ff7f35c7319cbef84246b/packages/grid/x-data-grid-pro/src/utils/tree/utils.ts#L268-L269

https://github.com/mui/mui-x/assets/13808724/e2f06815-92ad-4316-8ebd-de7d5ce6d1d4